pkg/monitor/instrumentation.go

```diff
@@ -0,0 +1,110 @@
+package monitor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Metrics struct {
+	EventCount                          prometheus.Gauge
+	Info                                *prometheus.GaugeVec
+	DurationCountByEventType            *prometheus.HistogramVec
+	DurationCountByCameraId             *prometheus.HistogramVec
+	SuccessRequest                      prometheus.Gauge
+	ErrorRequest                        prometheus.Gauge
+	StoreEventDataSuccess               prometheus.Gauge
+	TotalEventCountByTypeSuccess        prometheus.Gauge
+	TotalEventCountByCameraIdSuccess    prometheus.Gauge
+	EventCountSummaryByCameraIdSuccess  prometheus.Gauge
+	EventCountSummaryByEventTypeSuccess prometheus.Gauge
+	SummaryByCameraIdSuccess            prometheus.Gauge
+	SummaryByEventTypeSuccess           prometheus.Gauge
+}
+
+func NewMetrics(promCamRegistry prometheus.Registerer) *Metrics {
+	m := &Metrics{
+		EventCount: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: "myapp",
+			Name:      "connected_devices",
+			Help:      "Number of currently connected devices.",
+		}),
+		Info: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: "myapp",
+			Name:      "info",
+			Help:      "Information about the Event Stats environment.",
+		},
+			[]string{"version"}),
+		DurationCountByEventType: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Namespace: "myapp",
+			Name:      "request_duration_seconds",
+			Help:      "Duration of the request.",
+			Buckets:   []float64{0.1, 0.15, 0.2, 0.25, 0.3},
+		}, []string{"status", "method"}),
+		DurationCountByCameraId: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Namespace: "myapp",
+			Name:      "request_duration_seconds_totalEventCountByCameraId",
+			Help:      "Duration of the request totalEventCountByCameraId",
+			Buckets:   []float64{0.1, 0.15, 0.2, 0.25, 0.3},
+		}, []string{"status", "method"}),
+		SuccessRequest: prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
Few metrics from here can be removed by adding few labels to other metrics, try to design metrics in such a way that it can be used for multiple purposes.

Reply:
updated the code to add labels